### PR TITLE
Admin Portal Review Journals 

### PR DESCRIPTION
**Pull request by tbatk**
In this PR, I integrated the backend to the frontend of the admin portal's review journal page. The frontend is somewhat complex,  but I'll do my best to explain what exactly is going on. To start, I've made some changes to the `
JournalEntry` component that is present on this page. Instead of having one button that says "Review," it has been split into two buttons, "Approve" and "Reject."
overview

On these buttons, the ID of the journal entry is stored. When the "Approve" button is pressed, the ID is  taken from the button, and a request is sent to the backend to update the entry's review status. If the "Reject" button is selected, a modal appears that warns the user that their actions will delete the entry. 
modal

This modal is not attached  to the `JournalEntry` component, and because of that, it has no way to get the entry ID from the "Reject" button. The best way I could come up with is to store the entry ID of the last pressed "Reject" button in a state. If the modal is closed, then the ID is cleared. There's probably a better way to do this, but we can look into this later. Within the modal, the user has the option to reject the entry they have selected, or to close the modal. If they select the checkbox, then the modal will no longer appear upon pressing the reject button, and instead a request will immediately be sent to the backend to delete the entry.

Regardless of the action done, there are two messages that can appear after the request to the backend has been made. If the action was successful, then this message will show:
success
If there was some sort of error, then this message will show:
error

The only things missing from the page at this point are a link to view the entire article, and a way to refresh the list after the action has occured. I will add in the links once Zavier has finished the frontend for viewing individual blog posts. Please let me know if I need to change anything.

### CODE REVIEW

File: pages/portal/journal/review.tsx [hunk 1]
```diff
@@ -2,49 +2,207 @@ import { NextPage,  NextPageContext } from "next";
...
+                    <div className="modalBody">
+                        <h1>Are you sure?</h1>
+                        <p>Continuing with this action will delete the entry permanently.</p>
+                        <input type='checkbox' onClick={toggleWarningDismissed}/>
+                        <span>Do not show this message again.</span>
+                        <div className="actionButtonContainer">
+                            <button type="submit" name="reject" className="actionButton actionButtonPrimary" onClick={handleEntryApproval}>Reject</button>
+                            <button className="actionButton actionButtonSecondary" onClick={() => {setIsRejecting(false); setRejectingID("")}}>Close</button>
+                        </div>
+                    </div>
```
Comment (VishalAiely) [inline pages/portal/journal/review.tsx:81]: I like the idea of the confirmation modal to make sure people don't accidentally reject the wrong one

File: server/actions/Contentful.ts [hunk 2]
```diff
@@ -207,4 +208,50 @@ export async function getJournalEntryByType(type: string){
...
+    } catch (error){
+        if(error) console.error(error);
+        return false;
+    }
+}
+/**
+* @param id The unique journal identifier given by contentful.
+* @returns True if successful or false if there's an error.
+*/
+export async function deleteJournalEntryById(id: string){
```
Comment (VishalAiely) [inline server/actions/Contentful.ts:237]: I think deleting on rejecting is probably best to save space on Contentful, so this part looks good

File: server/actions/Contentful.ts [hunk 3]
```diff
@@ -128,6 +127,7 @@ export async function getJournalEntriesByReviewStatus(reviewed: boolean){
             journalEntry.id = entries.items[i].sys.id; // unique id for the asset
             journalEntry.body = entries.items[i].fields.body['en-US'];
             journalEntry.category = entries.items[i].fields.category['en-US'];
+            journalEntry.reviewed=  entries.items[i].fields.reviewed['en-US'];
```
Comment (VishalAiely) [inline server/actions/Contentful.ts]: minor fix to the spacing on this line as well as line 170

**Review by VishalAiely — APPROVED** [review]
I like the UI for the review list and the process seems to be smooth for the user. I'm glad you implemented the `deleteJournalEntry` to remove on rejections. Storing the entryID in a state when the modal is pressed is a fine solution to passing it between the reject button and the rejection handler.
Overall the code is well written, really good job!

File: server/actions/Contentful.ts (hunk 2 above)
Comment (tbatk) [reply server/actions/Contentful.ts:237]: All the credit goes to Kemal, he did a majority of the backend work for journal entries. Wouldn't have been possible without his help

File: components/Portal/JournalEntry/index.tsx [hunk 4]
```diff
@@ -1,25 +1,29 @@
 import styles from "./journalentry.module.scss";
-import {EntryProp} from "contentful-management/dist/typings/entities/entry";
```
Comment (kemalfidan) [inline components/Portal/JournalEntry/index.tsx:2]: Didn't realize there was a frontend component for this already - changing the return type probably messed this function up. Thanks for fixing it though!

**Review by kemalfidan — APPROVED** [review]

**Review by rluberto — APPROVED** [review]
Everything looks good to me! I like the new banner message that tells the user if something went wrong.

**Pull request merged**